### PR TITLE
bindinfo: add status vars for 'last_plan_binding_update_time'

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close [issue#26199](https://github.com/pingcap/tidb/issues/26199)

Problem Summary:
Add status vars for 'last_plan_binding_update_time'. We can use `show status like 'last_plan_binding_update_time';` to see the last plan binding update time in the server. 

### What is changed and how it works?
Add status variable to record the last plan binding update time.

### Check List 

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- Details of the last plan binding update time are now available in the last_plan_binding_update_time status variables

### CODE REVIEW

File: bindinfo/stat.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// GetScope gets the status variables scope.
+func (h *BindHandle) GetScope(status string) variable.ScopeFlag {
+	return variable.ScopeSession
+}
+
+// Stats returns the server statistics.
+func (h *BindHandle) Stats(vars *variable.SessionVars) (map[string]interface{}, error) {
+	m := make(map[string]interface{})
+	m[lastPlanBindingUpdateTime] = h.bindInfo.lastUpdateTime.String()
```
Comment (Reminiscent) [inline bindinfo/stat.go:32]: Do we need to lock here if there is concurrency here? @morgo

Comment (morgo) [reply bindinfo/stat.go:32]: I believe it is safe, since each concurrent request will generate its own map.

**Comment by morgo** [comment]
Please fix the linter issues, but otherwise this is looking good:
```
[2021-07-19T06:59:11.974Z] linting

[2021-07-19T06:59:12.232Z]   ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var update_time should be updateTime  

[2021-07-19T06:59:12.232Z]   bindinfo/bind_test.go:2130:2

[2021-07-19T06:59:12.232Z] 

[2021-07-19T06:59:12.232Z]   ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var update_time1 should be updateTime1  

[2021-07-19T06:59:12.232Z]   bindinfo/bind_test.go:2132:2

[2021-07-19T06:59:12.232Z] 

[2021-07-19T06:59:12.232Z]   ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var update_time2 should be updateTime2  

[2021-07-19T06:59:12.233Z]   bindinfo/bind_test.go:2135:2

[2021-07-19T06:59:12.233Z] 

[2021-07-19T06:59:12.233Z] ✘ 3 problems (3 errors, 0 warnings)

[2021-07-19T06:59:12.233Z] 

[2021-07-19T06:59:12.233Z] Errors:

[2021-07-19T06:59:12.233Z]   3  var-naming  
```

**Comment by Reminiscent** [comment]
@morgo Update, PTAL. Besides, @eurekaka thinks that the name `stat.go` will be ambiguous with the naming of statistics in the optimizer. What do you think?

**Review by morgo — APPROVED** [review]

File: bindinfo/bind_test.go [hunk 2]
```diff
@@ -2114,3 +2115,38 @@ func (s *testSuite) TestTemporaryTable(c *C) {
...
+	updateTimeStr := strings.Split(updateTimeVal, ".")
+
+	// The format of update time in stats var is 'xxxx-xx-xx xx:xx:xx'. And it will convert to 'xxxx-xx-xx xx:xx:xx.000'.
+	// So when we compare, we remove the last three digits and compare.
+	rows1 := tk.MustQuery("show status like 'last_plan_binding_update_time';").Rows()
+	updateTime1 := rows1[0][1]
+	updateTimeVal1, ok := updateTime1.(string)
+	c.Assert(ok, Equals, true)
+	updateTimeStr1 := strings.Split(updateTimeVal1, ".")
+	c.Assert(updateTimeStr1[0], Equals, updateTimeStr[0])
```
Comment (Reminiscent) [inline bindinfo/bind_test.go:2143]: This test is unstable. I'll fix it ASAP.

**Review by rebelice — APPROVED** [review]

File: bindinfo/bind_test.go (hunk 2 above)
Comment (eurekaka) [reply bindinfo/bind_test.go:2143]: Why will it be unstable?

File: bindinfo/stat.go (hunk 1 above)
Comment (eurekaka) [reply bindinfo/stat.go:32]: Use `h.bindInfo.Lock()` please, as what we do in `func (h *BindHandle) Update()`.

File: bindinfo/bind_test.go (hunk 2 above)
Comment (Reminiscent) [reply bindinfo/bind_test.go]: Update, the unstable test is fixed. Previous I use the `show global binding` to get the last update time. But the time format is not the same here(I have no idea why this happens in the test. If we use the MySQL Client to execute the same query we will get the same time format).

**Comment by Reminiscent** [comment]
@eurekaka Update. PTAL

**Review by eurekaka — APPROVED** [review]

**Pull request merged**